src/libraries/System.Net.Http/src/System/Net/Http/Headers/AltSvcHeaderParser.cs

```diff
@@ -73,10 +73,10 @@ private AltSvcHeaderParser()
             int? maxAge = null;
             bool persist = false;
 
-            while (idx != value.Length)
+            while (idx < value.Length)
```

Comment:
Same for many of these.

Reply:
I think these help code readability, let me know if I should revert.